ui/js/models/resultsets_store.js

```diff
@@ -289,19 +289,41 @@ treeherder.factory('ThResultSetStore', [
         };
 
         var addRunnableJobs = function(repoName, resultSet) {
-            return ThRunnableJobModel.get_list(repoName).then(function(jobList) {
-                var id = resultSet.id;
-                _.each(jobList, function(job) {
-                    job.result_set_id = id;
-                    job.id = thAggregateIds.escape(job.result_set_id + job.ref_data_name);
-                });
-
-                if (jobList.length === 0) {
-                    resultSet.isRunnableVisible = false;
-                    thNotify.send("No new jobs available");
+            // first check whether the Gecko Decision Task has completed
+            var tcURLPromise;
+            resultSet.platforms.forEach(function(platform) {
```

Comment:
The ``underscore`` library is pretty efficient.  You can replace the for loop with:


    var platform = _.find(resultSet.platforms, {
        "name": "gecko-decision",
        "groups": [{"jobs": [{"state": "completed"}]}]});